File: microservices/helium-transaction-client/src/main/java/es/caib/helium/transaction/config/JmsConfig.java
package es.caib.helium.transaction.config;

import com.fasterxml.jackson.databind.ObjectMapper;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.jms.annotation.EnableJms;
import org.springframework.jms.support.converter.MappingJackson2MessageConverter;
import org.springframework.jms.support.converter.MessageConverter;
import org.springframework.jms.support.converter.MessageType;

@Configuration
@EnableJms
public class JmsConfig {

//    @Bean
//    public DefaultJmsListenerContainerFactory jmsContainerFactory() {
//        DefaultJmsListenerContainerFactory containerFactory = new DefaultJmsListenerContainerFactory();
//        containerFactory.setPubSubDomain(true);
//        return containerFactory;
//    }

    @Bean
    public MessageConverter jacksonJmsMessageConverter(ObjectMapper objectMapper) {
        var converter = new MappingJackson2MessageConverter();
        converter.setTargetType(MessageType.TEXT);
        converter.setTypeIdPropertyName("_type");
        converter.setObjectMapper(objectMapper);
        return converter;
    }
}

File: microservices/helium-transaction-service/src/main/kotlin/es/caib/helium/transaction/config/JmsConfig.kt
package es.caib.helium.transaction.config

import com.fasterxml.jackson.databind.ObjectMapper
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.jms.support.converter.MappingJackson2MessageConverter
import org.springframework.jms.support.converter.MessageConverter
import org.springframework.jms.support.converter.MessageType

//@Configuration
class JmsConfig {

//    @Bean
//    fun jacksonJmsMessageConverter(objectMapper: ObjectMapper): MessageConverter {
//        val converter = MappingJackson2MessageConverter()
//        converter.setTargetType(MessageType.TEXT)
//        converter.setTypeIdPropertyName("_type")
//        converter.setObjectMapper(objectMapper)
//        return converter
//    }
}